cli: upgrade file filters in debug.zip

aa-joshi opened the pull request:
Previously, file filters in debug zip were extended to profiles & logs. This was inadequate because we collect other information where we would like to extend file filters. This change extends `include-files` & `exclude-files` file filters to
all grpc invocations. It will skip grpc invocations for the respective information if we have filtered out the file using flags.
This would improve debug zip performance in terms of memeory & time.

Epic: None
Part of: CRDB-39070
Release note(cli change): extends `include-files` & `exclude-files` flags for debug zip artifact collection.

aa-joshi committed:
cli: refactor zip_per_node.go

This file contains the grpc invocations to fetch & write observability
artifacts for debug.zip. This information is per node. We are improving file
filters for better usability of debug.zip as part of CRDB-39070. This patch
groups grpc invocations in methods so that subsequent filter changes are
applied per method.

Epic: none
Part of: CRDB-39070
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136098)
<!-- Reviewable:end -->

aa-joshi committed:
cli: add file filters for debug zip files

Previously, file filters in debug zip were extended to profiles & logs. This
was inadequate because we collect other information which is cluster wide as
well as node specific. This change extends `include-files` & `exclude-files`
file filters to all grpc invocations as part of debug zip artifacts. It will
skip grpc invocation for the respective information if we have filtered out the
output file using flags.  This would improve debug zip performance in terms of
memeory & time.

Epic: None
Part of: CRDB-39070
Release note(cli change): extends include-files & exclude-files flags per
node data collections. sample command: debug zip debug.zip --redact --insecure
--include-files="*" --exclude-files="*.log"

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @dhartunian, and @xinhaoz)

___
*[`pkg/cli/zip_helpers.go` line 304 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqj0_i2YaCgdI5vCxz:-OCqj0_i2YaCgdI5vCy-:bfcyjjw) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_helpers.go#L304)):*
> ```Go
> // shouldIncludeFile determine whether the given file name is included in the selection based on
> // include & exclude patterns.
> func (fs *fileSelection) shouldIncludeFile(filename string) bool {
> ```

NIT: I think this functionality is repeated in `isIncluded`? Maybe we should just call `shouldIncludeFile` inside `isIncluded`? To avoid duplication

https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_helpers.go#L269-L285
___
*[`pkg/cli/zip_per_node.go` line 44 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqtQTf3rO2Ev4bpmOr:-OCqtQTf3rO2Ev4bpmOs:b2nhk2c) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_per_node.go#L44)):*
> ```Go
> const gossipFileName = "gossip.json"
> const statusFileName = "status.json"
> const cpuProfileFileName = "cpu.pprof"
> ```

NIT



_Suggestion:_
```Go
const (
	regexpOfRemoteAddress   = "[[:alnum:].:-]+"
	stacksFileName          = "stacks.txt"
	stacksWithLabelFileName = "stacks_with_labels.txt"
	heapPprofFileName       = "heap.pprof"
	lsmFileName             = "lsm.txt"
	rangesInfoFileName      = "ranges.json"
	detailsFileName         = "details.json"
	gossipFileName          = "gossip.json"
	statusFileName          = "status.json"
	cpuProfileFileName      = "cpu.pprof"
)
```
___
*[`pkg/cli/zip_test.go` line 1367 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCquV8pB_sSnW0g-QoZ:-OCquV8pB_sSnW0g-Qo_:bz0t560) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_test.go#L1367)):*
> ```Go
> 	defer log.Scope(t).Close(t)
> 
> 	tests := []struct {
> ```

This looks like an anti-pattern. I might be wrong. 
Datadriven tests are suppose to replace table test. IIUC, each test case in this list has the file containing the expected output and the input to generate that output. 

The datadriven test framework allows you to give inputs/args/etc from within the file itself. And one file can contain multiple test cases. It also gives you utility functions to walk the dir containing these files. So, in that way it emulates table test.

This is the closest thing there is to documentation. Worth a read: 
https://github.com/cockroachdb/datadriven/blob/3a8210a831b8ad98e5daeed54ce6ff627f433c70/datadriven.go#L68-L114 
___
*[`pkg/cli/zip_cluster_wide.go` line 234 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqsrug1t1TtkJTOHKl:-OCqsrug1t1TtkJTOHKm:bjiu823) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_cluster_wide.go#L234)):*
> ```Go
> 				sLocality := zc.clusterPrinter.start(redact.Sprintf("writing tenant ranges for locality: %s", locality))
> 				name := fmt.Sprintf("%s/%s/%s", zc.prefix, tenantRangesName, locality)
> 				if !zipCtx.files.shouldIncludeFile(locality + ".json") {
> ```

If we use `name` instead, the user can also skip the entire tenant range right? `prefix_rangename_.*`? or even `prefix_.*`



_Suggestion:_
```Go
if !zipCtx.files.shouldIncludeFile(name + ".json") {
```
___
*[`pkg/cli/zip.go` line 521 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqjoS0Ea7V2DjGjPyg:-OCqjoS0Ea7V2DjGjPyh:bdih5a2) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L521)):*
> ```Go
> 	ctx := context.Background()
> 	fileName := sanitizeFilename(table)
> 	baseName := base + "/" + fileName
> ```

NIT: `path.Join` handles accidental trailing `/` and other edge cases. 



_Suggestion:_
```Go
baseName := path.Join(base, fileName)
```
___
*[`pkg/cli/zip.go` line 524 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqlt_j6gm-jy2IO6SE:-OCqlt_j6gm-jy2IO6SF:bdwgrfa) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L524)):*
> ```Go
> 	fileNameWithExtension := fileName + "." + zc.clusterPrinter.sqlOutputFilenameExtension
> 	if !zipCtx.files.shouldIncludeFile(fileNameWithExtension) {
> 		s := zr.start(redact.Sprintf("skipping table data for %s due to file filters", table))
> ```

Should this be `zr.info` instead? since this is not an incremental thing.


<!-- Sent from Reviewable.io -->

dhartunian reviewed (commented):
I review commit-by-commit in a multi-commit PR. In the future either squash stuff down if you revise prior work or give the reviewer a heads up. I was surprised to encounter a refactor of code I had already reviewed.

> Reviewed 1 of 1 files at r1, 3 of 3 files at r2, 1 of 1 files at r3, 1 of 14 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-:-ODq8vjc1QfmAwnNcslF:b-5aznn6)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, and @xinhaoz)

___
*[`-- commits` line 27 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-ODq6C0tBzAvdaPqPn-6:-ODq6C0tBzAvdaPqPn-7:beioevh):*
this release note does not contain enough information to explain the new usage. consider adding an example.
___
*[`-- commits` line 43 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-ODq8jrADk27iyqk1t6B:-ODq8jrADk27iyqk1t6C:b1shhbl):*
same commit message as prior commit? do they need to be squashed?
___
*[`pkg/cli/zip_test.go` line 1367 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCquV8pB_sSnW0g-QoZ:-ODqAXtq6qB4YKXBrSX6:b-1xbcb2) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_test.go#L1367)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

This looks like an anti-pattern. I might be wrong. 
Datadriven tests are suppose to replace table test. IIUC, each test case in this list has the file containing the expected output and the input to generate that input. 

The datadriven test framework allowes you to give inputs/args/etc from with the file. And one file can contain multiple tests cases. It also gives you utility functions to walk the dir containing these files. So, in that way it emulates table test.

This is the closest thing to documentation: 
https://github.com/cockroachdb/datadriven/blob/3a8210a831b8ad98e5daeed54ce6ff627f433c70/datadriven.go#L68-L114
</blockquote></details>

Agree with @arjunmahishi. Mixing tabledriven with datadriven is confusing. The test case must be readable and present in the datadriven tests, otherwise their meaning is obscured. The power of putting the test inputs + outputs together is that you can easily audit changes in both together.

It's straightforward to use datadriven to support something like the example below. There are many existing examples in the repository. Then all your test cases in the table can sit in 1 datadriven test file.

```

zip include=*.log,*.json exclude=*.log
----
<output>

```
___
*[`pkg/cli/zip_per_node.go` line 150 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-ODqALn45032u04zbFw1:-ODqALn45032u04zbFw2:bg7szw6) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_per_node.go#L150)):*
> ```Go
> 				return status.Gossip(ctx, &serverpb.GossipRequest{NodeId: id, Redact: zipCtx.redact})
> 			},
> 			pathName: prefix + "/gossip",
> ```

Can we need to lift the `gossip` and `details` names here into the constant?


<!-- Sent from Reviewable.io -->

aa-joshi committed:
cli: add test for debug zip filters
This change add test cases for the `include-files` & `exclude-files` flags
which are extended for cluster wide & node wide artifact collection.

Epic: none
Part of: CRDB-39070
Release note: None

aa-joshi reviewed (commented):
I had fixed few bugs which were found during dev testing & included same in last commit of tests. I had mentioned in commit message. However, I'll take note of it during next PRs. Thanks for the suggestion!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-:-OE-H_M94wmn2BtLsbZw:b-igj058)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi, @dhartunian, and @xinhaoz)

___
*[`-- commits` line 27 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-ODq6C0tBzAvdaPqPn-6:-OE-IHkNECLLTQO_zpO1:bb6tyiv):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

this release note does not contain enough information to explain the new usage. consider adding an example.
</blockquote></details>

DONE
___
*[`-- commits` line 43 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-ODq8jrADk27iyqk1t6B:-OE-IJXs438ROYrv1UCK:brcxrlo):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

same commit message as prior commit? do they need to be squashed?
</blockquote></details>

I had earlier kept 2 separate commits, one for per node data & other for cluster wide. So there were minimal changes in commit messages. I have squashed 2 commits now.
___
*[`pkg/cli/zip.go` line 521 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqjoS0Ea7V2DjGjPyg:-ODzPZJd3dUr9QpbhYM3:bb6tyiv) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L521)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

NIT: `path.Join` handles accidental trailing `/` and other edge cases.
</blockquote></details>

DONE
___
*[`pkg/cli/zip.go` line 524 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqlt_j6gm-jy2IO6SE:-OE-IddDAo8C6uVJT-Nu:b-jy3i21) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L524)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

Should this be `zr.info` instead? since this is not an incremental thing.
</blockquote></details>

`zr.info` doesn't support `RedactableString` as argument. Since it is table name & we need to strip markers, I am using here `zr.start` instead of `zr.info`. 
___
*[`pkg/cli/zip_cluster_wide.go` line 234 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqsrug1t1TtkJTOHKl:-OE-IlCM0DFOl3CIFk1e:bqq139a) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_cluster_wide.go#L234)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

If we use `name` instead, the user can also skip the entire tenant range right? `prefix_rangename_.*`? or even `prefix_.*`
</blockquote></details>

We use filepath.Match function to match the regex.  To match with the entire tenant range you need to provide regex something like `prefix*/*.json`. The file path tries to match with the exact path. Hence, we need to just check with last fileName. This change would make only sure of file filtering. If we need to skip directories then we might need to make some additional changes & introduce flags accordingly.
sample ex: https://go.dev/play/p/cuGfCcsuf22
___
*[`pkg/cli/zip_helpers.go` line 304 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqj0_i2YaCgdI5vCxz:-ODzVpAc3DvJ9XGR5H50:b-6hae7) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_helpers.go#L304)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

NIT: I think this functionality is repeated in `isIncluded`? Maybe we should just call `shouldIncludeFile` inside `isIncluded`? To avoid duplication

https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_helpers.go#L269-L285
</blockquote></details>

Thanks for pointing it out! I missed to stage changes.
___
*[`pkg/cli/zip_per_node.go` line 150 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-ODqALn45032u04zbFw1:-ODzXTaa14zyss1uwndE:bb6tyiv) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_per_node.go#L150)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Can we need to lift the `gossip` and `details` names here into the constant?
</blockquote></details>

DONE
___
*[`pkg/cli/zip_test.go` line 1367 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCquV8pB_sSnW0g-QoZ:-OE-Iva91YJXiDxBVPLT:bdscw0g) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip_test.go#L1367)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Agree with @arjunmahishi. Mixing tabledriven with datadriven is confusing. The test case must be readable and present in the datadriven tests, otherwise their meaning is obscured. The power of putting the test inputs + outputs together is that you can easily audit changes in both together.

It's straightforward to use datadriven to support something like the example below. There are many existing examples in the repository. Then all your test cases in the table can sit in 1 datadriven test file.

```

zip include=*.log,*.json exclude=*.log
----
<output>

```
</blockquote></details>

I referred the existing wrong pattern for writing the tests. Corrected. Thanks!


<!-- Sent from Reviewable.io -->

dhartunian reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, and @xinhaoz)

___
*[`pkg/cli/zip.go` line 524 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqlt_j6gm-jy2IO6SE:-OEQ_W892WL1PBxkds7_:b-xmu4yi) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L524)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

`zr.info` doesn't support `RedactableString` as argument. Since it is table name & we need to strip markers, I am using here `zr.start` instead of `zr.info`.
</blockquote></details>

If the `zipReporter` API does not support redaction correctly, please change it so that it does instead of using it incorrectly.

Also, I don't understand the phrase "Since it is table name & we need to strip markers". If we strip markers then we get rid of redactable output, right? Do we want to keep it? I'm confused.

Looking at the datadriven tests for debug zip, I don't see any redaction markers.
___
*[`pkg/cli/testdata/zip/file-filters/testzip_file_filters` line 2 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OEQcuUu33QfZopS0k6P:-OEQcuUu33QfZopS0k6Q:b-ar3p6d) ([raw file](https://github.com/cockroachdb/cockroach/blob/342cc0308700113381d186d653950cc821b6680e/pkg/cli/testdata/zip/file-filters/testzip_file_filters#L2)):*
> ```
> #include all filters
> --include-files=*
> ```

all the tests in here appear to use either `*` or `*.<ext>` match types. Can you add some cases that filter some more specific names. For instance, I could imagine someone omitting all the table dumps with `statistics` in the name or something.
___
*[`pkg/cli/testdata/zip/file-filters/testzip_file_filters` line 289 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OEQcflf2f4CXrNlrkjF:-OEQcflf2f4CXrNlrkjG:bfitqsi) ([raw file](https://github.com/cockroachdb/cockroach/blob/342cc0308700113381d186d653950cc821b6680e/pkg/cli/testdata/zip/file-filters/testzip_file_filters#L289)):*
> ```
> debug/hot-ranges.sh
> debug/pprof-summary.sh
> debug/tenant_ranges.err.txt
> ```

why does exclude still have some output?
___
*[`pkg/cli/testdata/zip/file-filters/testzip_file_filters` line 415 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OEQcls44BRlIov3Aqcp:-OEQcls44BRlIov3Aqcq:bx2u060) ([raw file](https://github.com/cockroachdb/cockroach/blob/342cc0308700113381d186d653950cc821b6680e/pkg/cli/testdata/zip/file-filters/testzip_file_filters#L415)):*
> ```
> --exclude-files=*.txt
> ----
> debug/debug_zip_command_flags.txt
> ```

why is this `txt` file included?


<!-- Sent from Reviewable.io -->

dhartunian reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi, @arjunmahishi, and @xinhaoz)

___
*[`pkg/cli/zip.go` line 524 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqlt_j6gm-jy2IO6SE:-OEQzZy_DrrEe169rtvr:bbk4c17) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L524)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

If the `zipReporter` API does not support redaction correctly, please change it so that it does instead of using it incorrectly.

Also, I don't understand the phrase "Since it is table name & we need to strip markers". If we strip markers then we get rid of redactable output, right? Do we want to keep it? I'm confused.

Looking at the datadriven tests for debug zip, I don't see any redaction markers.
</blockquote></details>

I took a closer look and it seems like `zr.start` supports redaction because of a change that modified `RunWithTimeout` to support redactable task names. The motivation was not because we need redaction in debug zip output. I think you can use `info` as-is and not worry about redaction here.


<!-- Sent from Reviewable.io -->

aa-joshi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi, @dhartunian, and @xinhaoz)

___
*[`pkg/cli/zip.go` line 524 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OCqlt_j6gm-jy2IO6SE:-OEXj18a9o3TEAF1XUZJ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4c1a0cbb05a6114956a759384e3e5e90117c563/pkg/cli/zip.go#L524)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

I took a closer look and it seems like `zr.start` supports redaction because of a change that modified `RunWithTimeout` to support redactable task names. The motivation was not because we need redaction in debug zip output. I think you can use `info` as-is and not worry about redaction here.
</blockquote></details>

Done.
___
*[`pkg/cli/testdata/zip/file-filters/testzip_file_filters` line 289 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OEQcflf2f4CXrNlrkjF:-OEXj4uV2a3RlI80GCrO:b2cs074) ([raw file](https://github.com/cockroachdb/cockroach/blob/342cc0308700113381d186d653950cc821b6680e/pkg/cli/testdata/zip/file-filters/testzip_file_filters#L289)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

why does exclude still have some output?
</blockquote></details>

Few files are generated for TSEs to run command on the debug.zip. We also don't want to filter out the err generated files. Hence, exclude all is more for the cluster generated specific files rather than all files inside debug.zip
___
*[`pkg/cli/testdata/zip/file-filters/testzip_file_filters` line 415 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-OEQcls44BRlIov3Aqcp:-OEXjUTY4Jmlq1DUpiun:b-vwrih8) ([raw file](https://github.com/cockroachdb/cockroach/blob/342cc0308700113381d186d653950cc821b6680e/pkg/cli/testdata/zip/file-filters/testzip_file_filters#L415)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

why is this `txt` file included?
</blockquote></details>

This is the file for support team to see all the flags which passed during command execution. The file filters are not applied on files or scripts which are used by TSEs.


<!-- Sent from Reviewable.io -->

arjunmahishi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136098#-:-OEYNzWoDezE51YWjqTm:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @dhartunian, and @xinhaoz)



<!-- Sent from Reviewable.io -->

aa-joshi commented:
TFTR!
bors r+

aa-joshi commented:
bors r+

aa-joshi commented:
bors r+
